spec/datadog/core/crashtracking/component_spec.rb

```diff
@@ -318,6 +375,82 @@
           end
         end
       end
+
+      describe 'Ruby and C method runtime stack capture' do
+        let(:runtime_stack) { crash_report_experimental[:runtime_stack] }
+
+        it 'captures both Ruby and C method frames in mixed stacks' do
+          expect_in_fork(fork_expectations: fork_expectations, timeout_seconds: 15) do
+            crash_tracker = build_crashtracker(agent_base_url: agent_base_url)
+            crash_tracker.start
+
+            ComponentSpecCrashStackHelper.new.top_level_ruby_method
+          end
+
+          frames = runtime_stack[:frames]
+          # Check that the crashing function is captured
+          expect(frames).to include(
+            hash_including(
+              function: 'free'
+            )
+          )
+
+          puts frames.to_json
+
+          # Sanity check some frames
+          expect(frames).to include(
+            hash_including(
+              function: 'ruby_method_with_c_calls'
+            )
+          )
+
+          expect(frames).to include(
+            hash_including(
+              function: 'top_level_ruby_method'
+            )
+          )
+
+          expect(frames).to include(
+            hash_including(
+              function: 'gsub'
+            )
+          )
+
+          expect(frames).to include(
+            hash_including(
+              function: 'each'
+            )
+          )
+        end
+      end
+    end
+
+    describe '#runtime_callback_registered?' do
+      it 'returns true when callback is registered' do
+        crashtracker = build_crashtracker(logger: logger)
+
+        expect(Datadog::Core::Crashtracking::RuntimeStacks).to receive(:_native_is_runtime_callback_registered).and_return(true)
+
+        expect(crashtracker.runtime_callback_registered?).to be true
+      end
+
+      it 'returns false when callback is not registered' do
+        crashtracker = build_crashtracker(logger: logger)
+
+        expect(Datadog::Core::Crashtracking::RuntimeStacks).to receive(:_native_is_runtime_callback_registered).and_return(false)
+
+        expect(crashtracker.runtime_callback_registered?).to be false
+      end
+
+      it 'returns false and logs debug when native method raises exception' do
+        crashtracker = build_crashtracker(logger: logger)
+        error = StandardError.new('Native error')
+
+        expect(Datadog::Core::Crashtracking::RuntimeStacks).to receive(:_native_is_runtime_callback_registered).and_raise(error)
+        expect(logger).to receive(:debug).with('Runtime stack callback status check not available: Native error')
+
+        expect(crashtracker.runtime_callback_registered?).to be false
+      end
     end
```

Comment:
I don't quite understand the need for `runtime_callback_registered?`... 

It's being added for this file only, and then we're testing it here but... we're using it nowhere? It seems like we could get rid of it instead?